pkg/config/proxy.go

```diff
@@ -28,64 +28,69 @@ const (
 )
 
 type Config struct {
-	Workdir string `yaml:"workdir"`
+	Workdir string `yaml:"workdir" toml:"workdir" json:"workdir"`
 
-	LCUrlsI []string  `yaml:"listen-urls"`
-	ACUrlsI []string  `yaml:"advertise-urls"`
-	LPUrlsI []string  `yaml:"listen-peer-urls"`
-	APUrlsI []string  `yaml:"advertise-peer-urls"`
-	LCUrls  []url.URL `yaml:"-"`
-	ACUrls  []url.URL `yaml:"-"`
-	LPUrls  []url.URL `yaml:"-"`
-	APUrls  []url.URL `yaml:"-"`
+	LCUrlsI []string  `yaml:"listen-urls" toml:"listen-urls" json:"listen-urls"`
+	ACUrlsI []string  `yaml:"advertise-urls" toml:"advertise-urls" json:"advertise-urls"`
+	LPUrlsI []string  `yaml:"listen-peer-urls" toml:"listen-peer-urls" json:"listen-peer-urls"`
+	APUrlsI []string  `yaml:"advertise-peer-urls" toml:"advertise-peer-urls" json:"advertise-peer-urls"`
+	LCUrls  []url.URL `yaml:"-" toml:"-" json:"-"`
+	ACUrls  []url.URL `yaml:"-" toml:"-" json:"-"`
+	LPUrls  []url.URL `yaml:"-" toml:"-" json:"-"`
+	APUrls  []url.URL `yaml:"-" toml:"-" json:"-"`
 
-	Config   ConfigManager `yaml:"config"`
-	Proxy    ProxyServer   `yaml:"proxy"`
-	API      API           `yaml:"api"`
-	Metrics  Metrics       `yaml:"metrics"`
-	Log      Log           `yaml:"log"`
-	Security Security      `yaml:"security"`
+	Config   ConfigManager `yaml:"config" toml:"config" json:"config"`
+	Proxy    ProxyServer   `yaml:"proxy" toml:"proxy" json:"proxy"`
+	API      API           `yaml:"api" toml:"api" json:"api"`
+	Metrics  Metrics       `yaml:"metrics" toml:"metrics" json:"metrics"`
+	Log      Log           `yaml:"log" toml:"log" json:"log"`
+	Security Security      `yaml:"security" toml:"security" json:"security"`
 }
 
 type Metrics struct {
-	PromCluster string `yaml:"prom_cluster"`
+	PromCluster string `yaml:"prom_cluster" toml:"prom_cluster" json:"prom_cluster"`
 }
 
 type ConfigManager struct {
-	IgnoreWrongNamespace bool `yaml:"ignore_wrong_namespace"`
+	IgnoreWrongNamespace bool   `yaml:"ignore_wrong_namespace" toml:"ignore_wrong_namespace" json:"ignore_wrong_namespace"`
+	WatchInterval        string `yaml:"watch_interval" toml:"watch_interval" json:"watch_interval"`
```

Comment:
I thought about this, it's hard to explain what this really means to the user.
For common cases, the config will be refreshed once updated. This interval is only used for defensive programming.
As for TiDB, the refresh interval for privileges and global variables are not exposed to users either, and nobody complains about it.

Reply:
We could categorize them into **KNOW WHAT YOU ARE DOING options** and provide a resonable default. If one don't understand, just don't set.

You can choose to use the default good config, or make your decision. That is users' freedom. BTW, these are possibly useful for debugging online.

Reply:
> As for TiDB, the refresh interval for privileges and global variables are not exposed to users either, and nobody complains about it.

There is no choice. If we have a documentation said the priv interval is XXX. Maybe someone will complain that it is too short or too long.

Reply:
> > As for TiDB, the refresh interval for privileges and global variables are not exposed to users either, and nobody complains about it.
> 
> There is no choice. They don't even know about that. Maybe they have a guess in their mind, and thought it is correct. If we have a documentation said the priv interval is XXX. Maybe someone will complain that it is too short or too long.

If we make it as a config, we must document it, and then users are confused and complain. I sometimes encounter the situation where a user saw a config and doesn't understand what it means and then asks us. It just brings us trouble. If we're removing it, it again brings compatibility problems.

Reply:
I would not consider it a problem. First of all, TiProxy will have much less options. It is much simpler and not that hard to understand.

My point is that documentation of complex options can be **highlighted**, in **another section** or **another page** with a **large NOTE**. The example for these options can be separated into **another file**, or **commented by default**.

If users feel normal options are enough, they surely won't dig deep into these complicated options. Moreover, we recommend you not to use these options unless you understand it. TiDB docs has none of these guidance. And description for system variables usually do not have any external link to the related page or doc.